### PR TITLE
Fix Nix build, add more Nix usage docs, more relaxed Rust build artifacts discovery

### PR DESCRIPTION
**Pull request by Pegasust**
- Should fix CI builds for Nix target
- Nix build should theoretically faster thanks to build artifact cache

### CODE REVIEW

**Comment by willruggiano** [comment]
I would actually prefer using https://github.com/yusdacra/nix-cargo-integration instead of crane but, alas, I'm not the one writing the code here :D

File: flake.nix [hunk 1]
```diff
@@ -51,7 +55,24 @@
...
+          inherit (self) meta;
+        };
+
+        packages.all = pkgs.callPackage ./default.nix {
+          sg-workspace = self.packages.${system}.workspace;
+          sg-plugin = self.packages.${system}.plugin;
+          inherit (self) meta;
+        };
+
+        packages.default = inputs.self.packages.${system}.all;
```
Comment (willruggiano) [inline flake.nix:75]: [`self'.packages.all`](https://flake.parts/module-arguments.html#self)

File: flake.nix [hunk 2]
```diff
@@ -51,7 +55,24 @@
 
         formatter = pkgs.alejandra;
 
-        packages.default = pkgs.callPackage ./. {inherit toolchain;};
+        packages.workspace = pkgs.callPackage ./workspace-drv.nix {
+          craneLib = inputs.crane.lib.${system}.overrideToolchain toolchain;
```
Comment (willruggiano) [inline flake.nix:59]: `inputs'.crane.lib.overrideToolchain`

File: flake.nix (hunk 1 above)
Comment (Pegasust) [reply flake.nix:75]: Done, thanks!

**Comment by Pegasust** [comment]
Not sure why MacOS is failing. I'll have to plug "it works on my machine" here. Maybe there's something wrong with `cachix/install-nix-action` itself?

**Comment by willruggiano** [comment]
> Not sure why MacOS is failing. I'll have to plug "it works on my machine" here. Maybe there's something wrong with `cachix/install-nix-action` itself?

Bumping the cachix action to v22 should fix it, according to the cachix docs

**Comment by willruggiano** [comment]
Noice

**Comment by Pegasust** [comment]
nci also looks very clean, I've decided to give it a try because the current build time is a bit too high (let's set up cachix at a later PR?). The TL;DR is nci doesn't support workspace quite too well yet https://github.com/yusdacra/nix-cargo-integration/issues/123. This is because of dream2nix.

**Comment by willruggiano** [comment]
Oh well, it'll catch up eventually. You might consider moving all of the *.nix files into a nix/ sub directory, to avoid cluttering the root with a bunch of contrib files. Up to @tjdevries

**Pull request merged**